torch/_dynamo/variables/torch.py

```diff
@@ -144,6 +144,75 @@ def _dynamo_overriden_transformers_eq(self, other):
     pass
 
 
+class TorchCtxManagerClassVariable(VariableTracker):
+    def __init__(self, value, **kwargs):
+        super().__init__(**kwargs)
+        self.value = value
+
+    def unique_var_name(self):
+        name = torch_get_name(self.value, f"allowed_fn_{id(self.value)}")
+        return "__" + re.sub(r"[^a-zA-Z0-9_]+", "_", name)
+
+    def reconstruct(self, codegen):
+        return codegen.setup_globally_cached(self.unique_var_name(), self.value, False)
+
+    def as_proxy(self):
+        return self.value
```

Comment:
You sure? Seems to me this should error...

Reply:
I copied from ```TorchVariable```. For a class, the proxy should be itself? like a constant?

Reply:
Oh, it should be unimplemented, will update it.